File: src/main/java/org/rakvag/blackjack/KortstokkProvider.java
package org.rakvag.blackjack;

import org.rakvag.blackjack.domene.Kortstokk;
import org.springframework.stereotype.Component;

//Poenget med denne klassen/Spring-bønna er å kunne mocke den i tester, slik at en kan returnere mock-kortstokker,
// som da gir støtte for å selv bestemme hvilke kort som returneres i testene
@Component
public class KortstokkProvider {

    public Kortstokk hentNyKortstokk() {
        return new Kortstokk();
    }

}

File: src/main/kotlin/org/rakvag/blackjack/KortstokkProvider.kt
package org.rakvag.blackjack

import org.rakvag.blackjack.domene.Kortstokk
import org.rakvag.blackjack.domene.KortstokkImpl
import org.springframework.stereotype.Component

@Component
class KortstokkProvider {

    fun hentNyKortstokk(): Kortstokk {
        return KortstokkImpl()
    }

}